File: bubble_sort_test.c
// gcc bubble_sort_test.c -o bubble_sort_test
//
// ./bubble_sort_test

#include <stdio.h>

void bubble_sort(int array[], int size) {
  for (int i = 0; i < size - 1; i++) {
    for (int j = 0; j < size - i - 1; j++) {
      if (array[j] > array[j + 1]) {
        int temp = array[j];
        array[j] = array[j + 1];
        array[j + 1] = temp;
      }
    }
  }
}

void print_array(int array[], int size) {
  printf("[");
  for (int i = 0; i < size; i++) {
    printf("%d", array[i]);
    if (i < size - 1) {
      printf(", ");
    }
  }
  printf("]");
}

int main() {
  int array[] = {64, 34, 25, 12, 22, 11, 90};

  printf("Original array:\n");
  print_array(array, sizeof(array) / sizeof(array[0]));
  printf("\n");

  bubble_sort(array, sizeof(array) / sizeof(array[0]));

  printf("\n");
  printf("Sorted array:\n");
  print_array(array, sizeof(array) / sizeof(array[0]));
  printf("\n");

  return 0;
}

File: BubbleSortTest.java
// javac BubbleSortTest.java
//
// java BubbleSortTest

public class BubbleSortTest {
    public static int[] bubbleSort(int[] array) {
        if (array != null) {
            for (int i = 0; i < array.length - 1; i++) {
                for (int j = 0; j < array.length - i - 1; j++) {
                    if (array[j] > array[j + 1]) {
                        int temp = array[j];
                        array[j] = array[j + 1];
                        array[j + 1] = temp;
                    }
                }
            }
        }
        return array;
    }

    public static String toString(int[] array) {
        if (array != null) {
            StringBuffer sb = new StringBuffer("");

            sb.append("[");
            for (int i = 0; i < array.length; i++) {
                sb.append(Integer.toString(array[i]));
                if (i < array.length - 1) {
                    sb.append(", ");
                }
            }
            sb.append("]");
            return sb.toString();
        } else {
            return null;
        }

    }

    public static void main(String[] args) {
        int[] array = { 64, 34, 25, 12, 22, 11, 90 };

        System.out.println("Original array:");
        System.out.println(toString(array));

        bubbleSort(array);

        System.out.println();
        System.out.println("Sorted array:");
        System.out.println(toString(array));
    }
}